schema/schema.go

```diff
@@ -835,17 +838,54 @@ func IsPreDefPredChanged(update *pb.SchemaUpdate) bool {
 	if !x.IsPreDefinedPredicate(update.Predicate) {
 		return false
 	}
-
```

Comment:
This function is not supposed to make changes to the schema. We need to break it down into two steps. First step, we should update the schema to what it should be. And then call this function "IsPreDefPredChanged" and not make any changes in the code of this function. See the code where it is being called from. It maybe useful to update the name of the function and do both steps in just one function so that it can be used in other places too if needed.

Reply:
i have changed function name to CheckAndModifyPreDefPredicate.